x-pack/elastic-agent/pkg/agent/control/server/listener.go

```diff
@@ -36,3 +39,8 @@ func createListener() (net.Listener, error) {
 	}
 	return lis, err
 }
+
+func cleanupListener() {
+	path := strings.TrimPrefix(control.Address(), "unix://")
+	os.Remove(path)
```

Comment:
same here, at least log error returned from remove, will make troubleshooting easier

Reply:
Same as above, will log it.